Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp

```diff
@@ -1142,7 +1142,9 @@ void NetworkConnectionToWebProcess::registerInternalBlobURLOptionallyFileBacked(
     CheckedPtr session = networkSession();
     if (!session)
         return;
-    if (blobFileAccessEnforcementEnabled() && shouldCheckBlobFileAccess())
+    // Skip file path checking for ephemeral sessions since blob files are in temporary directories
+    // that persist across WebContent process restarts but file access permissions don't.
+    if (blobFileAccessEnforcementEnabled() && shouldCheckBlobFileAccess() && !session->sessionID().isEphemeral())
```

Comment:
Instead of completely skipping the check, we might validate the requested url can only be the IDB temp blob url (matching some pattern) if it's ephemeral session.